app/actions/local/reactions.test.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import DatabaseManager from '@database/manager';
+import * as recentReactionsQueries from '@queries/servers/system';
+import * as emojiHelpers from '@utils/emoji/helpers';
+import * as logUtils from '@utils/log';
+
+import {addRecentReaction} from './reactions';
+
+import type ServerDataOperator from '@app/database/operator/server_data_operator';
+
+jest.mock('@database/manager');
+jest.mock('@queries/servers/system');
+jest.mock('@utils/emoji/helpers');
+jest.mock('@utils/log');
+
+describe('addRecentReaction', () => {
+    let operator: ServerDataOperator;
+    const serverUrl = 'baseHandler.test.com';
+
+    beforeEach(async () => {
+        jest.clearAllMocks();
+
+        await DatabaseManager.init([serverUrl]);
+        operator = DatabaseManager.serverDatabases[serverUrl]!.operator;
+
+        (recentReactionsQueries.getRecentReactions as jest.Mock).mockResolvedValue([]);
+        (emojiHelpers.getEmojiFirstAlias as jest.Mock).mockImplementation((emoji) => emoji);
+
+        operator.handleSystem = jest.fn();
+    });
+
+    afterEach(async () => {
+        await DatabaseManager.destroyServerDatabase(serverUrl);
+    });
+
+    it('should return an empty array if emojiNames is empty', async () => {
+        const result = await addRecentReaction(serverUrl, []);
+        expect(result).toEqual([]);
+    });
+
+    it('should add new emoji to the beginning of the list', async () => {
```

Comment:
This is not really testing what it says it should test.

Since there are no emojis in the list previously, I don't know if the expected value put them at the beginning or at the end.